app/models/list.model.js

```
@@ -0,0 +1,69 @@
+/**
+ * (c) 2016 Tieto Finland Oy
+ * Licensed under the MIT license.
+ */
+'use strict';
+
+/**
+* @ngdoc service
+* @name dashboard.ListData
+* @description
+* # ListData
+* Model in the dashboard.
+*/
+angular.module('dashboard')
+    .factory('ListData', ['AttachmentData', function (AttachmentData) {
+
+        function ListData(title, objects) {
+            this.title = title;
+            this.objects = objects;
+        }
+
+        ListData.createAttachmentList = function (title, attachment) {
+            if (typeof title === 'string') {
+                var array = [];
+                for (var index = 0; (attachment instanceof Array) && (index < attachment.length); index++) {
+                    var element = attachment[index];
+                    var item = AttachmentData.create(element.attachmentTitle, element.link, element.publicity, element.buttonType);
+                    if (item) {
+                        array.push(item);
+                    }
+                }
+                return new ListData(title, array);
+            }
+            return null;
+        };
+
+        ListData.createDecisionList = function (title, decision) {
+            if (typeof title === 'string') {
+                var array = [];
+                for (var index = 0; (decision instanceof Array) && (index < decision.length); index++) {
+                    var element = decision[index];
+                    var item = AttachmentData.create(element.decisionTitle, element.link, element.publicity, element.buttonType);
+                    if (item) {
+                        array.push(item);
+                    }
+                }
+                return new ListData(title, array);
+            }
+            return null;
```

Add an error log print for bad args cases, otherwise they're silently ignored.